framework/src/utils/MooseUtils.C

```diff
@@ -31,10 +31,44 @@
 
 // System includes
 #include <sys/stat.h>
+#include <numeric>
 
 namespace MooseUtils
 {
 
+// this implementation is copied from
+// https://en.wikibooks.org/wiki/Algorithm_Implementation/Strings/Levenshtein_distance#C.2B.2B
```

Comment:
Ok for the record, this is terrible C++11. It's cross between old C and C++11:
https://xkcd.com/1513/